src/dswx_sar/pre_processing.py

```diff
@@ -276,59 +376,81 @@ def run(cfg):
     # create instance to relocate ancillary data
     ancillary_reloc = AncillaryRelocation(ref_filename, scratch_dir)
 
-    # Check if the interpolated water body file exists
-    wbd_interpolated_path = Path(
-        os.path.join(scratch_dir, 'interpolated_wbd.tif'))
-
-    if not wbd_interpolated_path.is_file():
-        logger.info('interpolated wbd file was not found')
-        ancillary_reloc.relocate(wbd_file,
-                                 'interpolated_wbd.tif',
-                                 method='near')
-
-    # Check if the interpolated DEM exists
-    dem_interpolated_path = \
-        Path(scratch_dir) / 'interpolated_DEM.tif'
-    dem_reprocessing_flag = False
-    if not dem_interpolated_path.is_file():
-        logger.info('interpolated dem : not found ')
-
-        if os.path.isfile(dem_file) or check_gdal_raster_s3(dem_file, raise_error=False):
-            logger.info('interpolated dem file was not found')
-            ancillary_reloc.relocate(dem_file,
-                                     'interpolated_DEM.tif',
-                                     method='near')
+    # Note : landcover should precede before reference water.
+    relocated_ancillary_filename_set = {
+        'dem': 'interpolated_DEM.tif',
+        'hand': 'interpolated_hand.tif',
+        'landcover': 'interpolated_landcover.tif',
+        'reference_water': 'interpolated_wbd.tif',
+    }
+
+    input_ancillary_filename_set = {
+        'dem': dem_file,
+        'hand': hand_file,
+        'landcover': landcover_file,
+        'reference_water': wbd_file,
+    }
+
+    landcover_label = get_label_landcover_esa_10()
+
+    for anc_type, anc_filename in relocated_ancillary_filename_set.items():
+        input_anc_path = input_ancillary_filename_set[anc_type]
```

Comment:
Bit less verbose
```suggestion
    # key: ancillary file type
    # value: (relocated file name, input file name)
    ancillary_filenames_set = {
        'dem': ('interpolated_DEM.tif', dem_file),
        'hand': ('interpolated_hand.tif', hand_file),
        'landcover': ('interpolated_landcover.tif', landcover_file),
        'reference_water': ('interpolated_wbd.tif', wbd_file)
    }

    landcover_label = get_label_landcover_esa_10()

    for anc_type, (relo_anc_filename, input_anc_type) in ancillary_filenames_set.items():
```

Reply:
Sorry @LiangJYu I did not know that you were reviewing the PR. Thanks for the comments. I'll change code in another PR.